### PR TITLE
Improve url_package invalid type errors

### PR DESCRIPTION
**Pull request by ejanalysis**
## What changed

- Replaced the assertion-style `stopifnot()` check for `url_package(type = ...)` with an explicit, user-facing error.
- The error reports the supplied invalid value, lists every accepted `type`, and points users to `?url_package`.
- Added focused assertions for the invalid value, the beginning and end of the valid-option list, the help pointer, and multi-value input.

## Why

The previous assertion message exposed the implementation condition but did not tell users which values were valid or where to find the function documentation.

## User impact

Invalid calls now produce guidance such as:

```
Invalid `type` value: "not-a-real-type". `type` must be one of: "code", ... "api". See ?url_package.
```

## Validation

- Sourced `R/url_package.R` and ran `testthat::test_file("tests/testthat/test-url_package.R")` — 65 passed, 0 failed/warned/skipped
- Manually verified the complete invalid-type error text
- `git diff --check`

The broader `devtools::test(filter = "url_package")` entry point was blocked during package startup because the isolated worktree does not contain EJAM's ignored local Arrow datasets and its fallback dataset download failed; the focused test file itself passed completely using the sourced implementation.

— Codex

### CODE REVIEW

File: R/url_package.R [hunk 1]
```diff
@@ -213,7 +213,21 @@ url_package <- function(
   # (get_full_url = FALSE) makes sense
   repo_types <- c("code", "ejamrepo", "ejscreenrepo", "apirepo", "data", "datarepo")
 
-  stopifnot(length(type) == 1, type %in% names(field_by_type))
+  valid_types <- names(field_by_type)
+  if (length(type) != 1 || is.na(type) || !(type %in% valid_types)) {
```
Comment (ejanalysis) [inline R/url_package.R:230]: This validation occurs after the legacy-domain compatibility block, so `url_package(character())` never reaches the intended `<empty>` error. `all(character(0) %in% ...)` is `TRUE`, `domain <- type[1]` becomes `NA`, and the earlier `if (all(domain == "github.com"))` throws `missing value where TRUE/FALSE needed` (plus an obsolete-parameter warning). I reproduced that behavior on this head. Please move the cardinality/type validation ahead of the compatibility block, or guard that block with `length(type) == 1L`, and add `character()` to the focused tests.

— Codex

**Review by ejanalysis — COMMENTED** [review]
Quick review found one invalid-input edge case in the new validation; I reproduced it on this exact head and left the actionable detail inline. Apart from that case, the error-message change is focused, and this head combines cleanly with Public-Environmental-Data-Partners/EJAM#503.

— Codex

File: R/url_package.R (hunk 1 above)
Comment (ejanalysis) [reply R/url_package.R:230]: Fixed in f088f2ab: guarded the legacy block with `length(type) == 1L` (validation stays AFTER the block so legacy `type="github.com"` still remaps to `"code"` before being validated), and added `url_package(character())` to the focused tests — it now produces the intended `Invalid `type` value: <empty>` error. Verified by running the test file (0 failures).

**Review by ejanalysis — COMMENTED** [review]
**Claude-fix follow-up review** — reviewed final head `08e2cb0f` and the resulting `development` integration.

The empty-input issue is fixed without breaking the legacy `github.com`/`github.io` compatibility path: the legacy block now has a scalar-length guard, and the focused tests cover `url_package(character())`. I independently reproduced the intended `Invalid `type` value: <empty>` error on the merged code. The fix is also preserved through the Config/EJAM migration. No new actionable issue found. GitHub reports no check runs on this PR.

— Codex
